pkg/ipfs/utils.go

```diff
@@ -60,3 +62,24 @@ func ParsePeersString(peers []string) ([]peer.AddrInfo, error) {
 
 	return res, nil
 }
+
+func CopyFile(sourcePath, destinationPath string) error {
+	sourceFile, err := os.Open(sourcePath)
+	if err != nil {
+		return err
+	}
+	defer closer.CloseWithLogOnError(sourcePath, sourceFile)
+
+	destinationFile, err := os.Create(destinationPath)
+	if err != nil {
+		return err
+	}
+	defer closer.CloseWithLogOnError(destinationPath, destinationFile)
+
+	_, err = io.Copy(destinationFile, sourceFile)
```

Comment:
Whilst copy is pretty robust, if you follow frrst's advice above, it'd probably be worth checking the write-count against the fstat size.